src/community/wps-download/src/main/java/org/geoserver/wps/gs/download/DownloadUtilities.java

```diff
@@ -253,11 +254,11 @@ static File findStyle(StyleInfo style) throws IOException {
             }
             // the SLD file is not public, most probably it is located under a workspace.
             // lets try to search for the file inside the same layer workspace folder ...
-            File workspaces = new File(loader.getBaseDirectory().getCanonicalPath(), "/workspaces");
-            styleFile = new File(new File(workspaces, style.getWorkspace().getName() + "/styles"),
-                    style.getFilename());
+            Resource workspaces = loader.get ("/workspaces");
+            styleFile = workspaces.get(style.getWorkspace().getName() + "/styles/" + style.getFilename());
```

Comment:
String concatenation to build paths is an antipattern because it is too easy to miss a separator. Could we have a builder something like Resource.get(String ...) to make code cleaner?